source/server/connection_handler_impl.cc

```diff
@@ -478,6 +479,33 @@ void ConnectionHandlerImpl::ActiveTcpListener::newConnection(
 
   auto transport_socket = filter_chain->transportSocketFactory().createTransportSocket(nullptr);
   stream_info->setDownstreamSslConnection(transport_socket->ssl());
+#ifdef WIN32
```

Comment:
Now that you've added `ListenerFactoryContext` to the original_dst filter config, I think this entire block can move to that filter (which is logically where it belongs).

I think you'll need to add a method to `ListenerFilterCallbacks`, but the implementation of that interface (`ActiveTcpSocket`) already has a `stream_info_`, so I think that will be easy.

Reply:
Makes sense, I like this change it feels neater that way. Fixed in 2843f25025763c74a05673d5c618d9f83ce72e1c